File: RESTFul/AAB/PHP/app1/lib/AAB/Snapi.php
<?php
namespace Att\Api\AAB;

final class Snapi
{
    private $_type;
    private $_uri;
    private $_preferred;

    public function __construct($type, $uri, $preferred)
    {
        $this->_type = $type;
        $this->_uri = $uri;
        $this->_preferred = $preferred;
    }

    public function getSnapiType()
    {
        return $this->_type;
    }

    public function getUri()
    {
        return $this->_uri;
    }

    public function isPreferred()
    {
        if (is_string($this->_preferred)) {
            $lpref = strtolower($this->_preferred);
            return $lpref == 'true';
        }

        return $this->_preferred;
    }

    public function toArray()
    {
        $arr = array();
        if ($this->getSnapiType() !== null)
            $arr['type'] = $this->getSnapiType();

        if ($this->getUri() !== null)
            $arr['uri'] = $this->getUri();

        if ($this->isPreferred() !== null)
            $arr['preferred'] = $this->isPreferred();

        return $arr;
    }

    public static function fromArray($arr)
    {
        $type = $arr['type'];
        $uri = $arr['uri'];
        $preferred = $arr['preferred'];
        
        return new Snapi($type, $uri, $preferred);
    }
}

?>

File: RESTFul/AAB/Java/app1/src/main/java/com/att/api/aab/service/Snapi.java
/* vim: set expandtab tabstop=4 shiftwidth=4 softtabstop=4: */

/*
 * ====================================================================
 * LICENSE: Licensed by AT&T under the 'Software Development Kit Tools
 * Agreement.' 2014.
 * TERMS AND CONDITIONS FOR USE, REPRODUCTION, AND DISTRIBUTIONS:
 * http://developer.att.com/sdk_agreement/
 *
 * Copyright 2014 AT&T Intellectual Property. All rights reserved.
 * For more information contact developer.support@att.com
 * ====================================================================
 */

package com.att.api.aab.service;

public final class Snapi {
    private final String type;
    private final String uri;
    private final Boolean preferred;

    public Snapi(String type, String uri, Boolean preferred) {
        this.type = type;
        this.uri = uri;
        this.preferred = preferred;
    }

    public Boolean isPreferred() {
        return preferred;
    }

    public String getType() {
        return type;
    }

    public String getUri() {
        return uri;
    }

}
